### ISSUE TITLE
Guideline how to set up local database

### ISSUE THREAD

**Issue by slateroni** [ogarcia/rockpass#2]
as an avid user of lesspass and its database, I had been long thinking to setup a local DB on my home-RasPi server. I do have a basic understanding of setting up a webservice but never managed to get the provided docker container to run. 
When I found out about rockpass, it seemed like a simple-to-use solution and I wanted to try it! The binary runs fine on my Ubuntu 20.04 desktop for now and I can see and access the database with sqlite on the command line. But I'm somehow missing on how to properly address the database and create a user? For daily use I'll rely on the browser plugin and Android app. After adding a user manually in the database with the password '$2b$10$jJXcsuftQI9PwF8eqZo4/ObfBGbc.nhLBpV49fC4qeLLeh/Uz0YzW', I changed the URL in the browser plugin to localhost:8000 but am not able to login.

Apologies if this is a stupid request but I fancy that a brief guideline on how to setup and migrate the lesspass DB from the central server to a local rockpass instance might be of interest to others as well. 

This far is how I got:
1. download profiles from lesspass DB with cmdline tool: `lesspass --save`
2. download rockpass binary, modify Rocket.toml and execute: this creates /tmp/rockpass.sqlite for test purpose
3. manually add a user to DB: `sqlite3 /tmp/rockpass.sqlite` -- then: `insert into users values (1, "user@example.com", "$2b$10$jJXcsuftQI9PwF8eqZo4/ObfBGbc.nhLBpV49fC4qeLLeh/Uz0YzW");`

my questions:
- what is the proper way to create a user in rockpass DB?
- how can I import lesspass DB profiles into the rockpass DB? would `lesspass --load profiles.json --url localhost:8000` work?
- accessing the rockpass DB with the app or browser plugin should work then, shouldn't it? No need for a frontend?

I'd be happy to aid formulating some instructions once I have it running on my RasPi. For now I'm stuck that my RaspiOS is still 32-bit and the rockpass binary 64-bit, so I will need to upgrade the server but this is another story ;) 

Thanks for your work and for sharing rockpass.

**Comment by ogarcia** [comment]
The best way to create new users is simply to do it with the client application itself (the browser plugin). You go to the login section and hit the **New to LessPass? Join now** button and then enter the username and password of your choice and hit **Register**. To do this you must start Rockpass temporarily with the `registration_enabled = true` option. Once you have registered the users you need you can set it to `false`.

A quick way to do a test is to simply run the following:
```sh
export ROCKET_PORT=8080
export ROCKET_REGISTRATION_ENABLED=true
export ROCKET_DATABASES='{rockpass = { url = ":memory:" }}'
rockpass
```
And then access from the browser plugin as I said before.

When you have done the tests you can put the DB in a definitive location. For example:
```sh
export ROCKET_PORT=8080
export ROCKET_REGISTRATION_ENABLED=false
export ROCKET_DATABASES='{rockpass = { url = "/var/lib/rockpass/rockpass.sqlite" }}'
rockpass
```
Note that the `/var/lib/rockpass/` directory must already exist and you must have permissions to write to it.

Of course you can use the configuration file if you are more comfortable instead of exporting variables.

To run Rockpass on the Raspberry you will need to install Rust:
```sh
curl https://sh.rustup.rs -sSf | sh
```
And then simply clone the repository and compile it:
```sh
git clone https://github.com/ogarcia/rockpass.git
cd rockpass
cargo build --release
```
If all goes well you will have the binary in `target/release/rockpass` :wink: 

Once you have everything working you can check out my [Lesspass client](https://github.com/ogarcia/lesspass-client), which will allow you to automatically export and import your passwords.
```sh
lesspass-client -s https://api.lesspass.com -u user -p pass password export profiles.json
lesspass-client -s http://localhost:8080 -u user -p pass password import profiles.json
```

**Comment by ogarcia** [comment]
@slateroni I just noticed that Guillaume Vincent has disabled the ability to create accounts from the extension (I guess so that no more accounts are created on his server). There is another very simple way to create an account:
1. Run Rockpass (for example in memory for make test).
    ```sh
    export ROCKET_PORT=8080
    export ROCKET_REGISTRATION_ENABLED=true
    export ROCKET_DATABASES='{rockpass = { url = ":memory:" }}'
    rockpass
    ```
2. With LessPass generate a password with the following settings. Site: lesspass.com; Login: The email address you use to authenticate (in my example is `test@example.com`); Master password: The password you use to authenticate; Remaining options: untouched. This will generate your password. in my case `Kd*k5i63iN$^z)?V`.
3. Make a `curl` to create the account. Change the values of my example to your own.
    ```sh
    curl -d '{"email": "test@example.com","password": "Kd*k5i63iN$^z)?V"}' http://127.0.0.1:8080/auth/users/
    ```
    Rockpass will answer something like this:`{"detail":"Created test@example.com user"}`
4. Now you can login with any client (browser plugin, mobile app...).

Let me know if it has worked correctly.

**Comment by ogarcia** [comment]
In response to your request I have improved the documentation in the [README][1]. I guess it will be much more accessible now.

[1]: https://github.com/ogarcia/rockpass#usage

**Comment by slateroni** [comment]
sorry for being slow to respond.

Your hints were **gold** and helped me to get rockpass running on my RPi. Indeed creating a user was one of the major hurdles I was stuck on before.

The only _issue_ I ended up with was that I didn't managed to run rockpass with systemd and a non-default location of the database. But no worries, /var/lib/rockpass works fine, probably I missed something with permissions.

Also great that you updated the README with more examples, this will give interested users valuable hints no matter how they want to set it up. I add below my notes, feel free to comment and/or add them somewhere if you like.

--- 
### In case someone finds this useful, a brief summary of how I set up rockpass on my RPi4:
1. install rust on RPi: `curl https://sh.rustup.rs -sSf | sh` and follow instructions.
2. clone rockpass and build: 
    ```
    git clone https://github.com/ogarcia/rockpass.git
    cd rockpass
    cargo build --release
    ```
3. setup rockpass with systemd
    ```
    sudo wget https://aur.archlinux.org/cgit/aur.git/plain/rockpass.conf?h=rockpass -O /etc/rockpass.conf
    sudo wget https://aur.archlinux.org/cgit/aur.git/plain/rockpass.service?h=rockpass -O /etc/systemd/system/rockpass.service
    sudo cp <your build-dir>/target/release/rockpass /usr/bin/.
    ```
    - uncomment options in /etc/rockpass.conf and modify to your needs.
    - keep **default location** for database /var/lib/rockpass/rockpass.sqlite, else it will not work. An empty database will be allocated on first run
    - generate password with site: _lesspass.com_; login: _EMAIL_; master password: _MASTERPW_ -> creates **_LOGINPW_**
    - create user with `curl -d '{"email": "EMAIL","password": "LOGINPW"}' http://127.0.0.1:8080/auth/users`
4. migrate database from lesspass.com to rockpass with _lesspass-client_
    - export profiles from lesspass.com: 
          `lesspass-client -s https://api.lesspass.com -u EMAIL -p LOGINPW password export profiles.json`
    - import profiles in local rockpass database
          `lesspass-client -s http://localhost:8080 -u EMAIL -p LOGINPW password import profiles.json`

#### optional: remote access
if you have a dyndns address that links up with your homeserver (behind your firewall) you may want to access the database by something like, `<server-IP>/rockpass`. This requires quirking your webserver with a reverse-proxy.

In my case with **lighttpd**:
- reverse-proxy from 192.168.1.1/rockpass to 192.168.1.1:8080, add to /etc/lighttpd.conf
    ```
    $HTTP["url"] =~ "^/rockpass" {
	proxy.server = ( "" => ( ( "host" => "127.0.0.1", "port" => "8080" )))
	proxy.header = ( "map-urlpath" => ( "/rockpass" => ""))
    }
    ```
- **Note**: you may **not** want to use a path like _/rockpass_ that can be easily found by others and lets them communicate with your database. Use something random, for example a 16 character alphanumeric string created by lesspass :smile:

**Comment by ogarcia** [comment]
Thanks!

Yes I see some points that I can include in readme.
- The point 1 is detailed [here](https://github.com/ogarcia/rockpass#installing-rust)
- The pont 2 [here](https://github.com/ogarcia/rockpass#installing-rockpass)
- I will add points 3 and 4 to the README which are interesting and are not there now.

On the subject of the configuration and location of the database, I have made some changes in the code to make it easier to configure it. Simply copy the `rockpass.toml.example` file as `/etc/rockpass.toml` and modify it as you want, that should make any parameter work. The file you downloaded from the AUR to `/etc/rockpass.conf` is no longer needed and I have also modified the systemd file to not use it.

Tomorrow I will detail all this in the README.
